### ISSUE TITLE
[BUG] PST: external node, no local descriptors provided

### ISSUE THREAD

**Issue by eviltik** [mooijtech/go-pst#26]
**Describe the bug**
After change the PST file path in the example, then go build, i got "external node, no local descriptors provided" error.

**To Reproduce**

I've used the same code provided in your example and test on 2 different PST file, same error.

**Expected behavior**

Should run without error 

**Stack trace**

```
Initializing...
panic: Failed to open PST file: go-pst: external node, no local descriptors provided
github.com/mooijtech/go-pst/v5/pkg.(*File).GetHeapOnNodeReaderFromHID
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/heap_on_node.go:131
github.com/mooijtech/go-pst/v5/pkg.(*File).GetHeapOnNodeReaderFromHNID
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/heap_on_node.go:122
github.com/mooijtech/go-pst/v5/pkg.NewPropertyReader
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/property_reader.go:45
github.com/mooijtech/go-pst/v5/pkg.(*PropertyContext).GetPropertyReader
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/property_context.go:54
github.com/mooijtech/go-pst/v5/pkg.(*File).GetNameToIDMap
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/name_to_id_map.go:187
github.com/mooijtech/go-pst/v5/pkg.NewFromReaderWithBTrees
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/file.go:132
github.com/mooijtech/go-pst/v5/pkg.NewFromReader
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/file.go:52
github.com/mooijtech/go-pst/v5/pkg.NewFromFile
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/file.go:47
main.main
	/home/eviltik/toto/main.go:18
runtime.main
	/usr/local/go/src/runtime/proc.go:250
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1594
github.com/mooijtech/go-pst/v5/pkg.NewPropertyReader
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/property_reader.go:48
github.com/mooijtech/go-pst/v5/pkg.(*PropertyContext).GetPropertyReader
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/property_context.go:54
github.com/mooijtech/go-pst/v5/pkg.(*File).GetNameToIDMap
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/name_to_id_map.go:187
github.com/mooijtech/go-pst/v5/pkg.NewFromReaderWithBTrees
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/file.go:132
github.com/mooijtech/go-pst/v5/pkg.NewFromReader
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/file.go:52
github.com/mooijtech/go-pst/v5/pkg.NewFromFile
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/file.go:47
main.main
	/home/eviltik/toto/main.go:18
runtime.main
	/usr/local/go/src/runtime/proc.go:250
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1594
github.com/mooijtech/go-pst/v5/pkg.(*File).GetNameToIDMap
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/name_to_id_map.go:190
github.com/mooijtech/go-pst/v5/pkg.NewFromReaderWithBTrees
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/file.go:132
github.com/mooijtech/go-pst/v5/pkg.NewFromReader
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/file.go:52
github.com/mooijtech/go-pst/v5/pkg.NewFromFile
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/file.go:47
main.main
	/home/eviltik/toto/main.go:18
runtime.main
	/usr/local/go/src/runtime/proc.go:250
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1594
github.com/mooijtech/go-pst/v5/pkg.NewFromReaderWithBTrees
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/file.go:135
github.com/mooijtech/go-pst/v5/pkg.NewFromReader
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/file.go:52
github.com/mooijtech/go-pst/v5/pkg.NewFromFile
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.0/pkg/file.go:47
main.main
	/home/eviltik/toto/main.go:18
runtime.main
	/usr/local/go/src/runtime/proc.go:250
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1594


goroutine 1 [running]:
main.main()
	/home/eviltik/toto/main.go:21 +0x239
```

**Platform (please complete the following information):**
- go-pst v5
- go version go1.19.2 linux/amd64

**Comment by mooijtech** [comment]
@eviltik Thank you for your report! Can you test the latest version (v5.0.1) and see if this solves this issue?
Go modules should pick it up: https://github.com/mooijtech/go-pst/releases/tag/v5.0.1

**Comment by eviltik** [comment]
For sure ! let me try

**Comment by eviltik** [comment]
Sorry but i'm not familiar enought with golang ... 

I've updated my main.go 

before :
```
	pst "github.com/mooijtech/go-pst/v5/pkg"
```

after:
```
    pst "github.com/mooijtech/go-pst/releases/tag/v5.0.1"`
```

then 
```
eviltik@debian:~/toto$ go build
main.go:10:2: no required module provides package github.com/mooijtech/go-pst/releases/tag/v5.0.1; to add it:
	go get github.com/mooijtech/go-pst/releases/tag/v5.0.1
```

then, i have this error
```
eviltik@debian:~/toto$ go get github.com/mooijtech/go-pst/releases/tag/v5.0.1
go: module github.com/mooijtech/go-pst@upgrade found (v1.0.0), but does not contain package github.com/mooijtech/go-pst/releases/tag/v5.0.1
```

**Comment by eviltik** [comment]
Ok, i think i have to good cmd line to install the package
```
go get github.com/mooijtech/go-pst/v5/pkg@v5.0.1
```

But need i change something in main.go too ? Got the same "no local descriptors provided" for now.

**Comment by mooijtech** [comment]
@eviltik No worries!

Your Go project (assuming you created a new one) should have a `go.mod` (think package.json) file where Go dependencies are defined. This go.mod should indicate requires `github.com/mooijtech/go-pst v5.0.1`

The original import `pst "github.com/mooijtech/go-pst/v5/pkg` is correct.

You can alternatively clone/download the repository and in the source code directory `pkg` you should see `doc_test.go` which you can modify with your PST filepath and in the command line navigate inside this pkg/ directory and run `go test` which will execute `doc_test.go`. Note that if you do this the code which writes attachments expects a "attachments" directory.

If you are sure go.mod indicates v5.0.1 and the issue persists I would need to be able to reproduce this with the PST for debugging, my email is info@mooijtech.com but I doubt you would want to share the PST file which is understandable.

Please let me know if this issue is solved

**Comment by eviltik** [comment]
Yep ... i think it's ok i mean i'm using 5.0.1. But error is the same.

```
Initializing...
Walking folder: ROOT_FOLDER
Walking folder: Début du fichier de données Outlook
panic: Failed to walk folders: go-pst: external node, no local descriptors provided
github.com/mooijtech/go-pst/v5/pkg.(*File).GetHeapOnNodeReaderFromHID
	/home/eviltik/go/pkg/mod/github.com/mooijtech/go-pst/v5@v5.0.1/pkg/heap_on_node.go:131
[ ... ]
```

Ty for you email addr. Let me think about it.

**Comment by eviltik** [comment]
Yep, fixed in 5.0.2 :+1:

**Comment by mooijtech** [comment]
Closed since this is solved in v5.0.2
